### PR TITLE
Add few ways to load config

### PR DESCRIPTION
**Pull request by kotyara1005**

### CODE REVIEW

File: sanic/config.py [hunk 1]
```diff
@@ -97,12 +102,53 @@ def from_object(self, obj):
             if key.isupper():
                 self[key] = getattr(obj, key)
 
+    def from_json(self, filename, silent=False):
+        """Updates the values in the config from a JSON file. This function
+        behaves as if the JSON object was a dictionary and passed to the
```
Comment (r0fls) [inline sanic/config.py]: were you still finishing this docstring?

Comment (r0fls) [inline sanic/config.py]: Add a newline after `"""` as we do that throughout the rest of the code.

File: sanic/config.py [hunk 2]
```diff
@@ -97,12 +102,53 @@ def from_object(self, obj):
...
+                obj = json.loads(json_file.read())
+        except IOError as e:
+            if silent and e.errno in (errno.ENOENT, errno.EISDIR):
+                return False
+            e.strerror = 'Unable to load configuration file (%s)' % e.strerror
+            raise
+        return self.from_mapping(obj)
+
+    def from_mapping(self, *mapping, **kwargs):
+        """Updates the config like :meth:`update` ignoring items with non-upper
```
Comment (r0fls) [inline sanic/config.py]: Add a newline after """

File: sanic/config.py [hunk 3]
```diff
@@ -97,12 +102,53 @@ def from_object(self, obj):
...
+            )
+        mappings.append(kwargs.items())
+        for mapping in mappings:
+            for (key, value) in mapping:
+                if key.isupper():
+                    self[key] = value
+        return True
+
     def load_environment_vars(self):
+        """Looks for any SANIC_ prefixed environment variables and applies
```
Comment (r0fls) [inline sanic/config.py]: Leave this as is, put back the newline after """

**Comment by r0fls** [comment]
Can you add docs demonstrating the two new ways to add config?

**Comment by r0fls** [comment]
@kotyara1005 looks great, just a few things regarding updating docs and docstring formatting.

**Comment by r0fls** [comment]
hmm i guess since you went MIA and because other parts are using the `"""docstring...``` notation (rather than a newline) then we can merge this for now... note one of the doctstrings is still apparently incomplete.

**Comment by 38elements** [comment]
I think it is necessary to add this to [docs/sanic/config.md](https://github.com/channelcat/sanic/blob/master/docs/sanic/config.md).

**Comment by 38elements** [comment]
I think there is no need to make it possible to create configuration files by multiple formats.

**Comment by 38elements** [comment]
I think there is little needs to make it possible to create configuration files by multiple formats.

**Comment by r0fls** [comment]
ok let's close this for now then unless we see support to reopen it.

**Pull request closed without merging**